pkg/cmd/cli/cmd/cmd.go

```diff
@@ -65,20 +70,8 @@ func (o *CmdOptions) Complete(f *osclientcmd.Factory, cmd *cobra.Command, args [
 	if cmd.Flags().Lookup("workerconfigdir") != nil {
 		o.WorkerConfigDir = kcmdutil.GetFlagString(cmd, "workerconfigdir")
 	}
-	if cmd.Flags().Lookup("masterconfig") != nil && kcmdutil.GetFlagString(cmd, "masterconfig") != "" {
```

Comment:
oshinko-core already scans for the presence of a named master config or worker config and generates an error, so we should let oshinko-core do that instead of checking early in the CLI code. The error will be generated before any objects are created.